src/transport/raw/PeerTCPParamsStorage.cpp

```diff
@@ -0,0 +1,339 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the Peer TCP Parameter Storage object that provides
+ *      APIs for storage and retireval of the TCP parameters for a given peer
+ *      node scoped to a fabric.
+ */
+
+#include <transport/raw/PeerTCPParamsStorage.h>
+
+#include <lib/support/Base64.h>
+#include <lib/support/SafeInt.h>
+
+namespace chip {
+namespace Transport {
+
+constexpr TLV::Tag PeerTCPParamsStorage::kSupportedTransportsTag;
+constexpr TLV::Tag PeerTCPParamsStorage::kMaxTCPMessageSizeTag;
+
+StorageKeyName PeerTCPParamsStorage::GetStorageKey(const ScopedNodeId & node)
+{
+    return DefaultStorageKeyAllocator::TCPPeerParams(node.GetFabricIndex(), node.GetNodeId());
+}
+
+CHIP_ERROR PeerTCPParamsStorage::FindByScopedNodeId(const ScopedNodeId & node, uint16_t & supportedTransports,
+                                                    uint32_t & maxTCPMessageSize)
+{
+    ReturnErrorOnFailure(LoadTCPParamsFromStorage(node, supportedTransports, maxTCPMessageSize));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PeerTCPParamsStorage::SaveTrackedNodesList(const TrackedNodesList & list)
+{
+    std::array<uint8_t, MaxNodesListSize()> buf;
+    TLV::TLVWriter writer;
+    writer.Init(buf);
+
+    TLV::TLVType arrayType;
+    ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Array, arrayType));
+
+    for (size_t i = 0; i < list.mSize; ++i)
+    {
+        TLV::TLVType innerType;
+        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, innerType));
+        ReturnErrorOnFailure(writer.Put(kFabricIndexTag, list.mNodes[i].GetFabricIndex()));
+        ReturnErrorOnFailure(writer.Put(kPeerNodeIdTag, list.mNodes[i].GetNodeId()));
+        ReturnErrorOnFailure(writer.EndContainer(innerType));
+    }
+
+    ReturnErrorOnFailure(writer.EndContainer(arrayType));
+
+    const auto len = writer.GetLengthWritten();
+    VerifyOrReturnError(CanCastTo<uint16_t>(len), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    ReturnErrorOnFailure(mTCPParamsStorage->SyncSetKeyValue(DefaultStorageKeyAllocator::TCPPeerList().KeyName(), buf.data(),
+                                                            static_cast<uint16_t>(len)));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PeerTCPParamsStorage::LoadTrackedNodesList(TrackedNodesList & list)
+{
+    std::array<uint8_t, MaxNodesListSize()> buf;
+    uint16_t len = static_cast<uint16_t>(buf.size());
```

Comment:
```suggestion
    const uint16_t len = static_cast<uint16_t>(buf.size());
```

Reply:
Maybe consider creating a Span (initializing it out of an array initializes both buffer and length for it)

Reply:
Keeping it as a buffer and a non-const buffer size aligns with API requirements of the subsequent SyncGetKeyValue() call.